kolibri/core/kolibri_plugin.py

```diff
@@ -3,49 +3,28 @@
 """
 from __future__ import absolute_import, print_function, unicode_literals
 
-from django.core.urlresolvers import reverse
-from django.utils.translation import ugettext as _
-from kolibri.plugins.base import KolibriFrontEndPluginBase, KolibriPluginBase
-from kolibri.plugins.hooks import FRONTEND_PLUGINS, NAVIGATION_POPULATE
+from django.core.urlresolvers import reverse_lazy
+from django.utils.translation import ugettext_lazy as _
+from kolibri.core.webpack.hooks import FrontEndCoreHook
+from kolibri.plugins.base import KolibriPluginBase
 
-
-def main_navigation():
-    """
-    Callback for the plugin below
-    :return: A list of nav menu items used in the NAVIGATION_POPULATE hook.
-    """
-    return [{
-        'menu_name': _("Start page"),
-        'menu_url': reverse('kolibri:index'),
-    }]
+from . import hooks
 
 
 class KolibriCore(KolibriPluginBase):
     """
     The most minimal plugin possible. Because it's in the core, it doesn't define ``enable`` or ``disable``. Those
     methods should never be called for this plugin.
     """
-    def hooks(self):
+    pass
 
-        return {
-            NAVIGATION_POPULATE: main_navigation
-        }
 
+class MainNavigationItem(hooks.NavigationHook):
+    label = _("Start page")
+    url = reverse_lazy('kolibri:index')
 
-class KolibriCoreFrontEnd(KolibriFrontEndPluginBase):
-    """
-    Plugin to handle
-    """
-    entry_file = "assets/src/kolibri_core_app.js"
-    external = True
-    core = True
-
-    def hooks(self):
-        return {
-            FRONTEND_PLUGINS: self._register_front_end_plugins
-        }
-
-PLUGINS = (
-    KolibriCore,
-    KolibriCoreFrontEnd,
-)
+
+class DefaultFrontEndAsset(FrontEndCoreHook):
+    unique_slug = "default_frontend"
+    src_file = "kolibri/core/assets/src/kolibri_core_app.js"
```

Comment:
My point would be: Relative to what exactly? That's why I want the full path relative to the `npm run build` command. That's where things seem to kick off. Having the Python application which is essentially unaware of the npm configuration and build-out configuration say stuff (that `src_file` property) based on assumptions makes me feel OCD'ish :)

> Making people write out longer paths just increases the risk of error.

Not exactly sure that's the only likely outcome.... having paths assembled on to of hidden (inexplicit) assumptions about such paths would be a bigger concern IMO. If the path is wrong, this certainly fails anyways, and it'll fail loudly during `npm run build`.

People will use this to build their own plugins from their own file system structures. I'm not sure if my own approach above might actually render too *little* control to external plugins that wanna build webpacks against Kolibri's core :)